fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryUpdatePlan.java

```diff
@@ -88,7 +88,11 @@ public PipelineOperation getPipelineOperation() {
     @Nonnull
     @Override
     public <M extends Message> CompletableFuture<FDBStoredRecord<M>> saveRecordAsync(@Nonnull final FDBRecordStoreBase<M> store, @Nonnull final M message) {
-        return store.saveRecordAsync(message, FDBRecordStoreBase.RecordExistenceCheck.ERROR_IF_NOT_EXISTS_OR_RECORD_TYPE_CHANGED);
+        if (isDryRun()) {
```

Comment:
Rather than adding an `isDryRun()` check here, I think we should update `saveRecordAsync` in `RecordQueryAbstractDataModificationPlan` to take a `dryRun` Boolean (or an `ExecuteProperties`). Then the `executePlan` method in the abstract class can pass in the correct value from the `ExecuteProperties` it's given. That way, we don't have to mutate the state of the plan object (which generally should be immutable, as it can be cached and shared across multiple threads). Also, it looks like `RecordQueryInsertPlan` doesn't honor the dry-run flag, which I think would become more apparent if the abstract method was modified.